src/main/java/com/pnudev/librarysystem/security/UserDetailsImpl.java

```diff
@@ -0,0 +1,55 @@
+package com.pnudev.librarysystem.security;
+
+import com.pnudev.librarysystem.enums.UserRole;
+import lombok.AllArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.List;
+
+
+@AllArgsConstructor
+public class UserDetailsImpl implements UserDetails {
+
+    private String email;
+    private String password;
+    private UserRole role;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return List.of(new SimpleGrantedAuthority(role.name()));
+    }
+
+    @Override
+    public String getPassword() {
+        return this.password;
+    }
+
+    @Override
+    public String getUsername() {
+        return this.email;
+    }
+
+    @Override
+    public boolean isAccountNonExpired() {
+        return true;
+    }
+
+    @Override
+    public boolean isAccountNonLocked() {
+        return true;
```

Comment:
Why is a single implementation returning constant values? 
What are those supposed to represent?